radixdlt-core/radixdlt/src/main/java/com/radixdlt/tree/PMTBranch.java

```diff
@@ -122,13 +147,116 @@ public void insertNode(
       handleNewRemainder(key, val, acc, represent, read);
     } else {
       throw new IllegalStateException(
-          String.format("Unexpected subtree: %s", commonPath.whichRemainderIsLeft()));
+          String.format(UNEXPECTED_SUBTREE_ERROR_MSG, commonPath.whichRemainderIsLeft()));
+    }
+  }
+
+  @Override
+  public void removeNode(
+      PMTKey key, PMTAcc acc, Function<PMTNode, byte[]> represent, Function<byte[], PMTNode> read) {
+    final PMTPath commonPath = PMTPath.findCommonPath(this.getKey(), key);
+    if (commonPath.whichRemainderIsLeft() == PMTPath.RemainingSubtree.NONE) {
```

Comment:
to be exact this is caused by empty key. And the NONE remaining remainders is the consequence. Maybe lets have explicit condition?

Reply:
Not sure if I get it, this is the same logic used in `insertNode`.